src/output-json-tls.c

```diff
@@ -210,6 +211,16 @@ static void JsonTlsLogJa3(JsonBuilder *js, SSLState *ssl_state)
     }
 }
 
+static void JsonTlsLogSCJA4(JsonBuilder *js, SSLState *ssl_state)
+{
+    if (ssl_state->client_connp.ja4 != NULL) {
+        uint8_t buffer[36];
+        /* JA4 hash has 36 characters */
+        SCJA4GetHash(ssl_state->client_connp.ja4, (uint8_t(*)[36])buffer);
```

Comment:
Do we really need the cast ?

Reply:
Yes, otherwise we'd get
```
  CC       output-json-tls.o
output-json-tls.c: In function ‘JsonTlsLogSCJA4’:
output-json-tls.c:221:51: warning: passing argument 2 of ‘SCJA4GetHash’ from incompatible pointer type [-Wincompatible-pointer-types]
  221 |         SCJA4GetHash(ssl_state->client_connp.ja4, buffer);
      |                                                   ^~~~~~
      |                                                   |
      |                                                   uint8_t * {aka unsigned char *}
In file included from rust.h:24,
                 from app-layer-events.h:30,
                 from app-layer-parser.h:28,
                 from output-json-tls.c:40:
./../rust/gen/rust-bindings.h:5023:46: note: expected ‘uint8_t (*)[36]’ {aka ‘unsigned char (*)[36]’} but argument is of type ‘uint8_t *’ {aka ‘unsigned char *’}

```
I guess we could remove the string length from the interface of `SCJA4GetHash` but I'd rather have that checked.

Reply:
This is strange from the compiler...

Reply:
That's gcc 12.2.0  🤷🏻  I'd just keep the cast in. I will replace the hardcoded length with a preprocessor constant.